ros/src/computing/perception/localization/lib/autoware_lidar_localizer/src/map_manager/map_manager.cpp

```diff
@@ -0,0 +1,315 @@
+  const int min_y = std::floor(min_y_m / separate_map_size_);
+  const int max_y = std::floor(max_y_m / separate_map_size_);
+
+  for (int i = min_x; i <= max_x; ++i) {
+    for (int j = min_y; j <= max_y; ++j) {
+      boost::shared_ptr<pcl::PointCloud<PointTarget>> map_region_tmp_ptr(new pcl::PointCloud<PointTarget>);
+      passThroughPointCloud<PointTarget>(map_ptr, map_region_tmp_ptr, i * separate_map_size_,
+                                         j * separate_map_size_, separate_map_size_);
+
+      if (map_region_tmp_ptr->points.empty()) {
+        continue;
+      }
+
+      std::string path = directory_path + "/pointcloud_map_" +
+                         std::to_string(i * separate_map_size_) + "_" +
+                         std::to_string(j * separate_map_size_) + ".pcd";
+      boost::shared_ptr<pcl::PointCloud<PointTarget>> load_cloud_ptr(new pcl::PointCloud<PointTarget>);
+
+      boost::system::error_code file_error;
+      const bool exist_file = boost::filesystem::exists(path.c_str(), file_error);
+      if (!exist_file || file_error) {
+        std::cout << "no exist " << path << std::endl;
+      }
+      else if (pcl::io::loadPCDFile(path.c_str(), *load_cloud_ptr) == -1) {
+        std::cout << "load failed " << path << std::endl;
+      }
+
+      if(!load_cloud_ptr->points.empty()) {
+          const auto need_points_size = map_region_tmp_ptr->points.size() + load_cloud_ptr->points.size();
+          if(map_region_tmp_ptr->points.capacity() < need_points_size) {
+            map_region_tmp_ptr->points.reserve(need_points_size);
+          }
+          *map_region_tmp_ptr += *load_cloud_ptr;
+      }
+
+      boost::shared_ptr<pcl::PointCloud<PointTarget>> map_region_filtered_tmp_ptr(new pcl::PointCloud<PointTarget>);
+      donwsamplePointCloud(map_region_tmp_ptr, map_region_filtered_tmp_ptr, save_map_leaf_size_);
+
+      pcl::io::savePCDFileBinary(path, *map_region_filtered_tmp_ptr);
+    }
+  }
+}
+
+template <class PointTarget>
+void MapManager<PointTarget>::loadAroundMap(const Pose &localizer_pose) {
+
+  merged_map_.points.clear();
+
+  const int x = std::floor(localizer_pose.x / separate_map_size_);
+  const int y = std::floor(localizer_pose.y / separate_map_size_);
+
+  for (int i = -1; i <= 1; ++i) {
+    for (int j = -1; j <= 1; ++j) {
+      std::string path = directory_path_ + "/merged/pointcloud_map_" +
+                         std::to_string((i + x) * separate_map_size_) + "_" +
+                         std::to_string((j + y) * separate_map_size_) + ".pcd";
+      loadMap(path);
+    }
+  }
+
+  mtx_.lock();
+  merged_map_ptr_ = merged_map_.makeShared();
+  mtx_.unlock();
+}
+
+template <class PointTarget>
+void MapManager<PointTarget>::loadMap(const std::string &path) {
+
+  boost::system::error_code file_error;
+  const bool exist_file = boost::filesystem::exists(path.c_str(), file_error);
+  if (!exist_file || file_error) {
+    std::cout << "no exist " << path << std::endl;
```

Comment:
ROS_ERROR. 
Because we do not know who is outputting

Reply:
@yukkysaito 
I do not want to depend on ROS for library as much as possible.
How about changing it as follows?
`std::cerr << "[autoware_lidar_localizer]: no exist" << path << std::endl;`

Reply:
If you use `std::err` then this will:
- Be interleaved with output from the node in an uncontrollable way
- Not be manageable using the ROS logging output tools

If you do not want to depend on ROS logging mechanisms, then you should at least be using something like log4cpp so the output is controllable.